src/app/user/user.service.ts

```diff
@@ -98,7 +101,8 @@ export class UserService {
       .get(`${this.usersUrl}/${encodeURIComponent(userId)}`, { headers: this.headers })
       .map(response => {
         return response.json().data as User;
-      });
+      })
+      .catch(this.catchRequestError);
```

Comment:
Can you add tests for each of these catches. We need tests for the failing path not just the happy path.

Reply:
Sure. Will add some more tests.